tools/project_manager/diff.php

```diff
@@ -200,27 +236,24 @@ function get_navigation(
     }
     $navigation_text .= "\n" . _("Jump to") . ": <select name='jumpto' onChange='$jump_to_js'>\n";
 
-    $query = "
-        SELECT image,
-            $L_round->user_column_name,
-            CAST($L_round->text_column_name AS BINARY) = CAST($R_round->text_column_name AS BINARY) AS is_empty_diff
-        FROM $project->projectid
-        ORDER BY image ASC
-    ";
-    $res = DPDatabase::query($query);
+    $diff_array = load_page_diff_array($project, $L_round, $R_round, $format == "remove");
+
     $prev_image = "";
     $next_image = "";
     $prev_from_proofer = "";
     $next_from_proofer = "";
     $got_there = false;
     // construct the dropdown; work out where previous and next buttons should take us
-    while ([$this_val, $this_user, $is_empty_diff] = mysqli_fetch_row($res)) {
+    foreach ($diff_array as $this_val => $diff_record) {
+        $this_user = $diff_record["username"];
+        $is_diff = $format == "remove" ? $diff_record["is_diff_without_formatting"] : $diff_record["is_diff"];
```

Comment:
If you wanted to be slightly cleverer you could push the ternary inside the index:
```$is_diff == $diff_record[$format == "remove" ? "is_diff_without_formatting": "is_diff"];```
or
```
$idx = $format == "remove" ? "is_diff_without_formatting": "is_diff";
$is_diff = $diff_record[$idx];
```